src/shared/Core/CommandContext.cs

```diff
@@ -78,7 +85,7 @@ public interface ICommandContext : IDisposable
     /// </summary>
     public class CommandContext : DisposableObject, ICommandContext
     {
-        public CommandContext(string appPath, string installDir)
+        public CommandContext(string appPath, string installDir, string[] argv, long applicationStartTime = 0)
```

Comment:
Defaulting to `0` feels wrong. Why don't we default to computing the current time at the point the constructor is called here?

Given the first thing that almost all of the Program.cs entry points do is `new CommandContext(...)` it would be neater.

```csharp
// Program.cs
public static void Main(string args[])
{
    var appPath = ApplicationBase.GetApplicationPath();
    using (var context = new CommandContext(appPath, args))
    {
        ...
    }
}

// CommandContext.cs
public CommandContext(string appPath, string[] argv, long? startTime = null)
{
    long appStartTime = startTime ?? DateTimeOffset.UtcNow.ToUnixTimeMicroseconds();

    // The rest of CommandContext setup
    ...
}
```

We could also move the `ApplicationBase.GetApplicationPath` call in to the constructor too, if `null` was passed.

```csharp
// Program.cs
public static void Main(string args[])
{
    using (var context = new CommandContext(args))
    {
        ...
    }
}

// CommandContext.cs
public CommandContext(string[] argv, string appPath = null, long? startTime = null)
{
    long appStartTime = startTime ?? DateTimeOffset.UtcNow.ToUnixTimeMicroseconds();
    ApplicationPath = appPath ?? GetApplicationPath();

    // The rest of CommandContext setup
    ...
}
```

The ability to pass params other than `argv` to the CommandContext would be left if we wanted to test using a real `CommandContext` (rather than the `TestCommandContext`) but with stable paths and timestamps.